middleware/RetryHandler.md

```diff
@@ -9,9 +9,16 @@ Provide a reusuable component that provides application developers with effectiv
 - Retried Requests should be equivalent to the initial request made.
 - A `retry-attempt` header SHOULD be added which contains a numeric value representing the retry number.  This value will be used for diagnostics and determining the effectiveness of the retry handler.
 - Retries MUST respect the `retry-after` response header if provided.
-- Where no `retry-after` header is provided by the server, an exponential backoff with random offset hueristic should be used to determine the retry delay.
-- Retries should be limited to a maximum value.
-- Only requests with payloads that are buffered/rewindable are supported.  Payloads with forward only streams will be have the responses returned without any retry attempt.
+- Retries should be limited by maximum elapsed time and maximum retry attempts
+- Where no `retry-after` header is provided by the server, perform retry with the retry handler specified by the library consumer, or fall back to the default handler implemented as exponential backoff handler.
+- Allow library consumer to set the parameters for the exponential backoff, and also provide sane defaults, e.g.:
+  - initial retry interval: 0.5 seconds
```

Comment:
In our first implementation in .net we chose 10 secs as the first retry period.  The period was increased by 2^ retrycount.  So the second retry is 20 secs, then 80, 160 up to a maximum of 10 retries.

https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/src/Microsoft.Graph.Core/Requests/RetryHandler.cs#L184

0.5 secs feels pretty short.  Also using the 1.5 multiplier would produce a linear back-off rather than exponential.   If a service has a request/min limit a client could end up making an additional 12 requests that would all fail before the minute expires.

One update to the requirements that I just added is that middleware behavior can be influenced by individual requests by setting parameters on a middleware control object.  This allows individual requests to override the default middleware behavior.  We haven't currently implemented this, but it should be coming soon.

I am curious to hear more about your experiences that led you to picking the short retry time.  I also need to talk to Graph service teams to get a better understanding of what rate limiting periods currently are in use.